### PR TITLE
duppy.0.7.1

### PR DESCRIPTION
**Pull request by toots**

### CODE REVIEW

File: packages/duppy/duppy.0.7.1/opam [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+build: [
+  ["./configure" "--prefix" prefix]
+  [make]
+]
+install: [
+  [make "install"]
+]
+remove: ["ocamlfind" "remove" "duppy"]
+depends: [
+  "camlp4"
```
Comment (kit-ty-kate) [inline packages/duppy/duppy.0.7.1/opam]: Does it require the runtime part of camlp4?

Comment (toots) [reply packages/duppy/duppy.0.7.1/opam]: Yes but it's actually an optional dependency.. Fixing..

**Comment by kit-ty-kate** [comment]
Ok this is a tricky problem:
```
#=== ERROR while installing duppy.0.7.1 =======================================#
# opam-version         1.2.2 (aa258ecc06d3aea5a67f442a4ffd23f2a457180b)
# os                   linux
# command              make
# path                 /home/opam/.opam/4.05.0/build/duppy.0.7.1
# compiler             4.05.0
# exit-code            2
# env-file             /home/opam/.opam/4.05.0/build/duppy.0.7.1/duppy-7-4c6a1c.env
# stdout-file          /home/opam/.opam/4.05.0/build/duppy.0.7.1/duppy-7-4c6a1c.out
# stderr-file          /home/opam/.opam/4.05.0/build/duppy.0.7.1/duppy-7-4c6a1c.err
### stdout ###
# [...]
# making ._ncdi/duppy.di from duppy.mli
# make[2]: 'libduppy_stubs.a' is up to date.
# ocamlopt.opt -thread -c -dtypes -g -w +A-4@5-7@8-9@11@12@20-35-44-45-50 -I /home/opam/.opam/4.05.0/lib/pcre -I /home/opam/.opam/4.05.0/lib/bytes duppy.ml
# ocamlopt.opt -thread -a     -I /home/opam/.opam/4.05.0/lib/pcre -I /home/opam/.opam/4.05.0/lib/bytes   -ccopt -L/home/opam/.opam/4.05.0/lib      -cclib -lduppy_stubs   \
# 			 -o duppy.cmxa duppy.cmx
# make[2]: Leaving directory '/home/opam/.opam/4.05.0/build/duppy.0.7.1/src'
# ocamlfind ocamlc -c -package camlp4 -package camlp4.quotations.o -package camlp4.extend -syntax camlp4o pa_duppy.mli
# Makefile:51: recipe for target 'pa_duppy.cmi' failed
# make[1]: Leaving directory '/home/opam/.opam/4.05.0/build/duppy.0.7.1/src'
# Makefile:12: recipe for target 'all' failed
### stderr ###
# >> Fatal error: OCaml and preprocessor have incompatible versions
# Fatal error: exception Misc.Fatal_error
# make[1]: *** [pa_duppy.cmi] Error 2
# make: *** [all] Error 2
```
It seems to have detected the camlp4 for the system compiler and not for the compiler that is actually used. Could you find a way to detect it correctly?

**Comment by toots** [comment]
Tricky indeed. The `camlp4` binary is indeed discarded as having an incompatible version. However, the `camlp4` module is detected. From what I can see, modules detections are done via `ocamlfind query <module>`. So, it seems to me that the issue is that `ocamlfind` picks up the system-wide `camlp4` package, which I'm not sure that I can do anything about.

I could try to push a test PR to check on this assumption. I have no idea how to interact/test things in the CI environment..

**Comment by toots** [comment]
Actually I can reproduce locally:
```
% opam switch 4.06.1
(...)
% eval `opam config env`
% opam install ocamlfind
(...)
% ocamlfind query camlp4
/Users/toots/.opam/4.06.1/lib/ocaml/camlp4
% ls /Users/toots/.opam/4.06.1/lib/ocaml/camlp4
ls: /Users/toots/.opam/4.06.1/lib/ocaml/camlp4: No such file or directory
zsh: exit 1     ls -GhH /Users/toots/.opam/4.06.1/lib/ocaml/camlp4
```

**Comment by toots** [comment]
I'm gonna investigate this. Meanwhile, I think I should just revert to making it a dependency..

**Comment by kit-ty-kate** [comment]
Ok fair enough. Thanks!

**Pull request merged**